### DISCUSSION TITLE
Add fragment deduplication to `client` preset

### DISCUSSION THREAD

**Discussion by nandorojo** [dotansimha/graphql-code-generator#8617] [General]
I think it would be awesome to add fragment deduplication to the client preset. I'm running into this issue quite a bit.

for example:

```graphql
fragment Page on Event {
  ...Content

  artists {
    id
    ...Artist
  }
}

fragment Content on Event {
  artists {
    ...Artist
  }
}

fragment Artist on Artist {
  id
}
```

Here, I get the following error: `There can be only one fragment named 'Artist'`

**Comment by saihaj** [comment]
I have spread a fragments multiple times. Can you share a reproduction for this?

**Reply by nandorojo** [reply]
Yeah sure, what's the best playground for the client preset for reproductions?

**Reply by charlypoly** [reply]
You can use our [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) available on [GitHub](https://github.com/dotansimha/graphql-code-generator-issue-sandbox-template), [Stackblitz](https://stackblitz.com/github/dotansimha/graphql-code-generator-issue-sandbox-template) or [CodeSandbox](https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template)

**Reply by lrholmes** [reply]
Hey 👋  I have also encountered this issue, and it's a bit of a potential blocker so I'm hopeful we can resolve it. I've created a minimal repro here: https://codesandbox.io/s/focused-silence-flkogf

It is a very minimal repro but I think shows the essence of the issue. If you run `npm run start`, the GraphQL query is printed showing two instances of the example `SharedComponentFragment` fragments I included. This is essentially how my query looks when seeing the error at the network level `There can only be one fragment named 'X'`, so hopefully serves as a suitable reproduction — let me know if not though!

**Reply by saihaj** [reply]
Thanks for the repro @lrholmes let's move discussion to this issue https://github.com/dotansimha/graphql-code-generator/issues/8670
